core/matchers/src/main/kotlin/au/com/dius/pact/core/matchers/MatcherExecutor.kt

```diff
@@ -119,10 +105,51 @@ fun <M : Mismatch> domatch(
             matchMaxType(matcher.max, path, expected, actual, mismatchFn)
     is IncludeMatcher -> matchInclude(matcher.value, path, expected, actual, mismatchFn)
     is NullMatcher -> matchNull(path, actual, mismatchFn)
+    is IgnoreOrderMatcher -> matchIgnoringOrder(path, expected, actual, mismatchFn)
+    is MinEqualsIgnoreOrderMatcher -> matchMinEquals(matcher.min, path, expected, actual, mismatchFn)
+    is MaxEqualsIgnoreOrderMatcher -> matchMaxEquals(matcher.max, path, expected, actual, mismatchFn)
+    is MinMaxEqualsIgnoreOrderMatcher -> matchMinEquals(matcher.min, path, expected, actual, mismatchFn) +
+            matchMaxEquals(matcher.max, path, expected, actual, mismatchFn)
     else -> matchEquality(path, expected, actual, mismatchFn)
   }
 }
 
+fun <M : Mismatch> matchIgnoringOrder(
+  path: List<String>,
+  expected: Any?,
+  actual: Any?,
+  mismatchFactory: MismatchFactory<M>
+): List<M> {
+  val matches = when {
+    actual == null && expected == null -> true
+    actual is JsonPrimitive && expected is JsonPrimitive -> {
+      logger.debug { "matchIgnoringOrder skipping primitive" }
+      true
+    }
+    actual is JsonArray && expected is JsonArray -> {
+      val actualMap = instanceMapOf(actual.map { it.toString() })
+      val expectedMap = instanceMapOf(expected.map { it.toString() })
+      when {
+        actualMap == expectedMap -> true
+        subsetOf(expectedMap, actualMap) -> true
+        expected.size() > actual.size() -> false
+        else -> {
+          val actualJsonObjects: List<JsonElement> = actual.filter { it is JsonElement }
+          val expectedJsonObjects: List<JsonElement> = expected.filter { it is JsonElement }
+          actualJsonObjects.isNotEmpty() || expectedJsonObjects.isNotEmpty()
+        }
+      }
+    }
+    else -> false
+  }
+  logger.debug { "comparing ${valueOf(expected)} to ${valueOf(actual)} ignoring order of elements at $path -> $matches" }
+  return if (matches) {
+    emptyList()
+  } else {
+    listOf(mismatchFactory.create(expected, actual, "Expected ${valueOf(expected)} to equal ${valueOf(actual)} ignoring order of elements", path))
+  }
```

Comment:
This seems duplicative of [`compareListContentUnordered`](https://github.com/mitre/pact-jvm/blob/a57de008f7a54d7d4ea78ca39d9da763854a6a39/core/matchers/src/main/kotlin/au/com/dius/pact/core/matchers/JsonBodyMatcher.kt#L108).

When I tried to change it to the below, more tests seemed to pass (with the exception changes to the error messages in [`'matching json bodies - return a mismatch - with ignore-order - when actual is missing an element'`](https://github.com/mitre/pact-jvm/blob/a57de008f7a54d7d4ea78ca39d9da763854a6a39/core/matchers/src/test/groovy/au/com/dius/pact/core/matchers/JsonBodyMatcherSpec.groovy#L366)).


```suggestion
  return when {
    actual is JsonArray && expected is JsonArray -> emptyList()
    else -> matchEquality(path, expected, actual, mismatchFactory)
  }
```

In addition to JsonArray, we would probably also need to cover cases like List and Element.